autoload/neomake/makers/ft/d.vim

```diff
@@ -49,8 +49,7 @@ function! neomake#makers#ft#d#dmd()
     return {
         \ 'args': l:args,
         \ 'errorformat':
-        \     '%-G%f:%s:,%f(%l): %m,' .
-        \     '%f:%l: %m',
+        \     '%f(%l\,%c): %m,'
```

Comment:
You've removed 3 errorformat entries and only added one. Are you sure you didn't mean to just replace the last one? Also, if the program may not always give a column number (even with the `-vcolums` argument - some errors may not be column specific), you should just add your errorformat entry instead of repalacing them all.